BitwardenShared/Core/Auth/Repositories/AuthRepository.swift

```diff
@@ -960,8 +960,19 @@ extension DefaultAuthRepository: AuthRepository {
         } else {
             account.profile.userId.hashColor
         }
+        let hasMasterPassword: Bool = await (
+            try? stateService.getUserHasMasterPassword(userId: account.profile.userId)
+        ) ?? false
+        let isUnlockWithPinOn: Bool = await (
+            try? stateService.pinProtectedUserKey(userId: account.profile.userId)
+        ) != nil
+        let isUnlockWithBiometricOn: Bool = await (
+            try? biometricsRepository.getBiometricUnlockStatus().isEnabled
+        ) ?? false
+        let canBeLocked = hasMasterPassword || isUnlockWithPinOn || isUnlockWithBiometricOn
```

Comment:
🤔 Does it make sense to extract this logic into a func in this repository? So it's easier to be reused if needed elsewhere. It can be private for the time being if you want. What do you think?
🤔 Additionally, I see these functions already being used which is alike the logic you have here but without the user id:
- `isPinUnlockAvailable()`
- `canVerifyMasterPassword()`
Would it make sense to improve these functions passing `userId: String?` and having an extension method with no parameter that passes `userId: nil` so we reuse them in this logic as well?
I feel like by reusing these functions, then if the state service changes somehow it would be less code to maintain/update.

Reply:
I think that is good improvement, I am sure we would need this ```canBeLocked``` check somewhere else. I will add it.